netket/graph/symmetry.py

```diff
@@ -80,6 +81,62 @@ def remove_duplicates(self, *, return_inverse=False):
         else:
             return group
 
+    def inverse(self):
+        """
+        Returns reordered SymmGroup where the each element is the inverse of
+        the original symmetry element. If g = self[element] and h = self.inverse()[element]
+        gh = I for all elements.
+
+        """
+
+        automorphisms = self.to_array()
+        n_symm = len(automorphisms)
+        inverse = np.zeros([n_symm], dtype=int)
+        automorphisms = np.array(automorphisms)
+        for i, perm1 in enumerate(automorphisms):
+            for j, perm2 in enumerate(automorphisms):
+                perm_sq = perm1[perm2]
+                if np.all(perm_sq == np.arange(len(perm_sq))):
+                    inverse[i] = j
+
+        return SymmGroup([self.elems[i] for i in inverse], self.graph)
+
+    def group_algebra(self):
```

Comment:
I'm not sure "group algebra" is a clear name for this function. My group theory knowledge is somewhat rusty, but (aside from using `g^-1`) doesn't it essentially compute a representation of the group's Cayley table (i.e., it's multiplication table)?

Or is there an established meaning of "group algebra" that coincides with what this function does? Otherwise I think we should try and find a different name.

Reply:
It's a Cayley table but one side is the involution of the group. I'm open to renaming it though.

Reply:
How about flattenen_Cayley(). And something like returns a flattened Cayley table, s.t. for u = flattenen_Cayley()[elem1*n_symm + elem2],  g = self.inverse()[elem1],  and h = self[elem2], u = gh

Reply:
actually it would be u = self[flattenen_Cayley()[elem1*n_symm + elem2]], maybe this is too cumbersome

Reply:
*flattened

Reply:
Now that you mention it: Is there a reason the flattened table is returned? It sounds more generally useful for me to return the `ndim=2` table. Flattening it when needed is easier than the reverse operation.

Reply:
Now that you mention it: Is there a reason the flattened table is returned? It sounds more generally useful for me to return the `ndim=2` table. Flattening it when needed is easier than the reverse operation.

Reply:
Okay I've gone ahead and changed the name to flattened_Cayley, and rewrote the description with that in mind

Reply:
Cayley table is I think a bit too much group theory jargon (especially given that it's not exactly the table of `g*h` due to the inverse) for the name. How about just `product_table`?

Reply:
product table sounds fine. I guess I could just return an array of n_dim = 2 and flatten/make it Hashable inside of setup()?

Reply:
Actually I'm just going to have DenseEquivariant take flattened_product_table and complain if it gets something else. GCNN will do the same if product_table is specified, and if it just gets SymmGroup

Reply:
> product table sounds fine. I guess I could just return an array of n_dim = 2 and flatten/make it Hashable inside of setup()?

How about this: The `DenseEquivariant` layer accepts `Union[HashableArray, SymmGroup]` and in setup you either use the array directly or, if the attribute is a `SymmGroup` call something like `group.product_table().ravel()` and work with that?

Then everything is consistent with `DenseSymm`.

Reply:
Okay great, I will do that